Examples/Swift/POIAlongRouteExample.swift

```diff
@@ -0,0 +1,149 @@
+        style.addSource(source)
+
+        // Create new layer for the line.
+        let layer = MGLLineStyleLayer(identifier: "polyline", source: source)
+
+        // Set the line join and cap to a rounded end.
+        layer.lineJoin = NSExpression(forConstantValue: "round")
+        layer.lineCap = NSExpression(forConstantValue: "round")
+
+        // Set the line color to a constant blue color.
+        layer.lineColor = NSExpression(forConstantValue: UIColor(red: 59/255, green: 178/255, blue: 208/255, alpha: 1))
+
+        // Use `NSExpression` to smoothly adjust the line width from 2pt to 20pt between zoom levels 14 and 18. The `interpolationBase` parameter allows the values to interpolate along an exponential curve.
+        layer.lineWidth = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)",
+                                       [14: 2, 18: 20])
+
+        // We can also add a second layer that will draw a stroke around the original line.
+        let casingLayer = MGLLineStyleLayer(identifier: "polyline-case", source: source)
+        // Copy these attributes from the main line layer.
+        casingLayer.lineJoin = layer.lineJoin
+        casingLayer.lineCap = layer.lineCap
+        // Line gap width represents the space before the outline begins, so should match the main line’s line width exactly.
+        casingLayer.lineGapWidth = layer.lineWidth
+        // Stroke color slightly darker than the line color.
+        casingLayer.lineColor = NSExpression(forConstantValue: UIColor(red: 41/255, green: 145/255, blue: 171/255, alpha: 1))
+        // Use `NSExpression` to gradually increase the stroke width between zoom levels 14 and 18.
+        casingLayer.lineWidth = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)", [14: 1, 18: 4])
+
+        // Just for fun, let’s add another copy of the line with a dash pattern.
+        let dashedLayer = MGLLineStyleLayer(identifier: "polyline-dash", source: source)
+        dashedLayer.lineJoin = layer.lineJoin
+        dashedLayer.lineCap = layer.lineCap
+        dashedLayer.lineColor = NSExpression(forConstantValue: UIColor.white)
+        dashedLayer.lineOpacity = NSExpression(forConstantValue: 0.5)
+        dashedLayer.lineWidth = layer.lineWidth
+        // Dash pattern in the format [dash, gap, dash, gap, ...]. You’ll want to adjust these values based on the line cap style.
+        dashedLayer.lineDashPattern = NSExpression(forConstantValue: [0, 1.5])
+
+        guard let poiLayer = self.mapView.style?.layer(withIdentifier: "poi-label") as? MGLSymbolStyleLayer else {
+            return
+        }
+        style.insertLayer(layer, below: poiLayer)
+        style.insertLayer(dashedLayer, above: layer)
+        style.insertLayer(casingLayer, below: layer)
+
+    }
+
+    func restrictPOIVisibleShape() {
```

Comment:
Perhaps add a comment explaining the steps in this function

Reply:
@pjleonard37 Good point. I updated this!